http.js

```diff
@@ -393,11 +392,11 @@ exports.read = function (request, qualifier) {
             if (Buffer.isBuffer(data)) {
                 contentType = response.headers['content-type'] || '',
                 match = contentType.match(/charset=([a-z0-9-]+)/i),
-                charset = match ? match[1] : 'utf8';
+                charset = match && iconv.encodingExists(match[1]) ? match[1] : 'utf8';
```

Comment:
:ok: Sounds good.

Frankly, threading charset through NodeReader and NodeWriter is a wart necessitated to take advantage of Node’s internal transcoding for UTF-8/UCS2/ASCII. I would have much preferred for that layer to just dish up Buffer and pipe through the appropriate decoder at a higher level. Even if we move in that direction, we should still allow V8 to transcode those three charsets if V8 is in fact faster than transcoding with iconv-lite.